File: android/app/src/main/java/com/mediadownloader/extractors/sites/InstagramExtractor.kt
package com.mediadownloader.extractors.sites

import com.mediadownloader.config.Config
import com.mediadownloader.extractors.BaseExtractor
import com.mediadownloader.extractors.DownloadTask

class InstagramExtractor : BaseExtractor {
    override fun matches(url: String): Boolean =
        url.contains("instagram.com")

    override suspend fun extract(url: String): List<DownloadTask> =
        listOf(
            DownloadTask(
                url = url,
                useYtdlp = true,
                formatId = Config.getDefaultFormat()
            )
        )
}

File: src/extractors/sites/instagram_extractor.cpp
#include "instagram_extractor.hpp"
#include "../../config.hpp"

bool InstagramExtractor::matches(
    const std::string& url
)
{
    return
        url.find("instagram.com")
        != std::string::npos;
}

std::vector<DownloadTask>
InstagramExtractor::extract(
    const std::string& url
)
{
    DownloadTask task;

    task.url = url;

    task.use_ytdlp = true;

    task.format_id =
        Config::get_default_format();

    return { task };
}
